domain/schema/model/sql/0022-resource.sql

```diff
@@ -29,103 +29,85 @@ INSERT INTO resource_state VALUES
 
 CREATE TABLE resource (
     uuid TEXT NOT NULL PRIMARY KEY,
-    application_uuid TEXT NOT NULL,
+    charm_uuid TEXT NOT NULL,
     name TEXT NOT NULL,
     revision INT,
     origin_type_id INT NOT NULL,
     state_id INT NOT NULL,
```

Comment:
**observation/suggestion:** The entries in `resource_state` are quite confusing to me. The difference between `'available'` vs `'potential'` is not immediately clear to me, the comment on about available is confusing. Is a resource always available or potential? If so, why not just have a boolean flag for potential that can be set to true if the resource has a new version available instead of a separate table with two states in it.

 Though I know its not changed in the PR so this can be looked at later.